erpnext/selling/doctype/sales_order/sales_order.js

```diff
@@ -226,10 +226,43 @@ erpnext.selling.SalesOrderController = erpnext.selling.SellingController.extend(
 						fields: fields,
 						primary_action: function() {
 							var data = d.get_values();
+
+							data_object_items = [];
+							for(i = 0; i < data["items"].length; i++) {
+								data_object_items.push(data.items[i].sales_order_item);
+							}
+
+							visible_rows = $(".modal-dialog:has(div[data-fieldtype=\"Table\"][data-fieldname=\"items\"]) .grid-body .grid-row");
```

Comment:
Why isn't there a better way to do this?!

Reply:
That jquery selector is fugly, this will match any hidden dialogs that might be on the dom and not exclusive to the open dialog.

Consider using:

`visible_rows = d.get_field("items").$wrapper.find(".grid-body .grid-row");`

or something similar. Remember to check /public/js/frappe/ui/dialog.js as there is an API there
dialog.js also inherits from field_group.js which has even more ways to reach fields.